custom_dc/custom_data.md

```diff
@@ -0,0 +1,291 @@
+
+You can have a multi-level group hierarchy by using `/` as a separator between each group.
+
+`earchDescriptions`
+
+: Formerly `nlSentences`. An array of descriptions to be used for creating more NL embeddings for the variable. This is only needed if the variable `name` is not sufficient for generating embeddings.  
+
+Note: `nlSentences` is deprecated and will be removed in the future.
+
+### Sources
+
+The `sources` section is optional. It encodes the sources and provenances associated with the input dataset. Each named source is a mapping of provenances to URLs. 
+
+#### Source parameters
+
+`url`
+: Required: The URL of the named source. For example, for named source `U.S. Social Security Administration`, it would be `https://www.ssa.gov`.
+
+`provenances`
+: Required: A set of name:URL pairs. Here are some examples:
+
+```json
+{
+  "USA Top Baby Names 2022": "https://www.ssa.gov/oact/babynames/"
+  "USA Top Baby Names 1923-2022": "https://www.ssa.gov/oact/babynames/decades/century.html"
+}
+```
+
+## Load local custom data
+
+To load custom data uploaded to Google Cloud, see instead [Pointing the local Data Commons site to the Cloud data](testing_cloud.md) for procedures.
+
+### Start the Docker container with local custom data {#docker-data}
+
+Once you have your CSV files and config.json set up, use the following command to restart the Docker container, mapping your custom data directory to the Docker userdata directory.
+
+<pre>
+docker run -it \  
+-p 8080:8080 \  
+-e DEBUG=true \  
+--env-file $PWD/custom_dc/sqlite_env.list \  
+-v $PWD/custom_dc/<var>CUSTOM_DATA_DIRECTORY</var>:/userdata \  
+[-v $PWD/custom_dc/<var>CUSTOM_DATA_DIRECTORY</var>/datacommons:/sqlite]  
+gcr.io/datcom-ci/datacommons-website-compose:stable  
+</pre>
+
+The optional `-v` flag preserves the SQLite data so it loads automatically when you restart the Docker container.
+
+Every time you make changes to the CSV or JSON files, you should reload the data, as described below.
+
+## Load custom data in SQLite
+
+As you are iterating on changes to the source CSV and JSON files, you will need to reload the data. Custom Data Commons allows you to reload data on the fly, while the website is running, so even multiple users can reload data with a shared Docker instance.
+
+You can load the new/updated data from using the /admin page on the site, or using curl, from the command line:
+
+1. Optionally, in the `sqlite_env.list` file, set the `ADMIN_SECRET` environment variable to a string that authorizes users to load data.
+1. Start the Docker container as usual, being sure to map the path to the directory containing the custom data (see command above). 
+1. With the services running, do either of the following:
+    -  In the browser, navigate to the `/admin page`. If a secret is required, enter it in the text field, and click **Load**. 
+    -  From a terminal window command line, run the following:
```

Comment:
I'm wondering if we should remove mention of this option? Soon enough, we'll have the standalone script and we'd want users to use that option over the web ones.

Reply:
Let's leave it until that's ready. I can update the doc then.